### PR TITLE
Add constructors for data that only validates size

### PR DESCRIPTION
**Pull request by coltfred**
Added constructors for the data types which only validate size in their `from_bytes`. This allows consumers who have already statically proven the size to avoid `.unwrap` calls that can never actually throw.

### CODE REVIEW

**Comment by coltfred** [comment]
@isislovecruft Would you be willing to give me some feedback on this PR?

**Comment by coltfred** [comment]
Ping! I'd love to have some feedback on this PR if at all possible.

File: src/ed25519.rs [hunk 1]
```diff
@@ -427,6 +433,19 @@ impl<'a> From<&'a SecretKey> for ExpandedSecretKey {
...
+
+    ///Construct an ExpandedSecretKey given an array of exactly `EXPANDED_SECRET_KEY_LENGTH` bytes.
+    fn new(bytes: [u8; EXPANDED_SECRET_KEY_LENGTH]) -> ExpandedSecretKey {
+        let mut lower: [u8; 32] = [0u8; 32];
+        let mut upper: [u8; 32] = [0u8; 32];
+
+        lower.copy_from_slice(&bytes[00..32]);
+        upper.copy_from_slice(&bytes[32..64]);
+
+        ExpandedSecretKey{ key:   Scalar::from_bits(lower),
```
Comment (isislovecruft) [inline src/ed25519.rs]: I don't trust my users to always hand me canonical bits in reduced form, and I don't want to provide them with constructors which they can use as foot cannons.

Comment (coltfred) [reply src/ed25519.rs]: I must be confused. https://github.com/dalek-cryptography/ed25519-dalek/blob/master/src/secret.rs#L401 provides a way to do exactly this code, but verifies the length. If we statically guarantee the length then the function is the same as `from_bytes` but without the failure condition. I can understand if you don't want to call it `new`. `from_sized_bytes` might be a better option.

**Pull request closed without merging**